code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java

```diff
@@ -152,7 +156,7 @@ private LdapContext login(String username, String password) {
             userContext = new InitialLdapContext(props, null);
             return userContext;
         } catch (NamingException e) {
-            logger.info("Failed to login to ldap user:" + username, e);
+            logger.info("Failed to login to ldap user:" + username + " " + e.getExplanation());
```

Comment:
Noticed that in some cases you re-throw NamingException, and in some cases you log an error and return null. If it depends on a context its called from, could you elaborate the use cases?